consensus/types_internal.go

```diff
@@ -0,0 +1,197 @@
+	aggregatedSignature AggregatedSignature
+}
+
+func (qc *quorumCertificate) GetBlockHash() BlockHash {
+	return qc.blockHash
+}
+
+func (qc *quorumCertificate) GetView() uint64 {
+	return qc.view
+}
+
+func (qc *quorumCertificate) GetAggregatedSignature() AggregatedSignature {
+	return qc.aggregatedSignature
+}
+
+//////////////////////////////////////////////////////////
+// AggregatedSignature interface implementation for testing
+//////////////////////////////////////////////////////////
+
+type aggregatedSignature struct {
+	signersList *bitset.Bitset
+	signature   *bls.Signature
+}
+
+func (as *aggregatedSignature) GetSignersList() *bitset.Bitset {
+	return as.signersList
+}
+
+func (as *aggregatedSignature) GetSignature() *bls.Signature {
+	return as.signature
+}
+
+//////////////////////////////////////////////////////////
+// Block interface implementation for testing
+//////////////////////////////////////////////////////////
+
+type block struct {
+	blockHash   BlockHash
+	height      uint64
+	view        uint64
+	qc          QuorumCertificate
+	aggregateQC AggregateQuorumCertificate
+}
+
+func (b *block) GetBlockHash() BlockHash {
+	return b.blockHash
+}
+
+func (b *block) GetHeight() uint64 {
+	return b.height
+}
+
+func (b *block) GetView() uint64 {
+	return b.view
+}
+
+func (b *block) GetQC() QuorumCertificate {
+	if !isInterfaceNil(b.aggregateQC) {
+		return b.aggregateQC.GetHighQC()
+	}
+	return b.qc
+}
+
+//////////////////////////////////////////////////////////
+// VoteMessage interface implementation for testing
+//////////////////////////////////////////////////////////
+
+type voteMessage struct {
+	view      uint64
+	blockHash BlockHash
+	publicKey *bls.PublicKey
+	signature *bls.Signature
+}
+
+func (vm *voteMessage) GetView() uint64 {
+	return vm.view
+}
+
+func (vm *voteMessage) GetBlockHash() BlockHash {
+	return vm.blockHash
+}
+
+func (vm *voteMessage) GetPublicKey() *bls.PublicKey {
+	return vm.publicKey
+}
+
+func (vm *voteMessage) GetSignature() *bls.Signature {
+	return vm.signature
+}
+
+//////////////////////////////////////////////////////////
+// TimeoutMessage interface implementation for testing
+//////////////////////////////////////////////////////////
+
+type timeoutMessage struct {
+	view      uint64
+	highQC    QuorumCertificate
+	publicKey *bls.PublicKey
+	signature *bls.Signature
+}
+
+func (tm *timeoutMessage) GetView() uint64 {
+	return tm.view
+}
+
+func (tm *timeoutMessage) GetHighQC() QuorumCertificate {
+	return tm.highQC
+}
+
+func (tm *timeoutMessage) GetPublicKey() *bls.PublicKey {
+	return tm.publicKey
+}
+
+func (tm *timeoutMessage) GetSignature() *bls.Signature {
+	return tm.signature
+}
+
+//////////////////////////////////////////////////////////
+// Internal blockWithValidatorLookup type
+//////////////////////////////////////////////////////////
+
+type blockWithValidatorLookup struct {
+	block           Block
+	validatorSet    []Validator          // Ordered slice of validators
```

Comment:
Nit: I would maybe call this validatorList rather than validatorSet. I know we normally talk about the validator set, but it confused me a bit because I thought this was a set struct.